networkx/exception.py

```diff
@@ -52,3 +52,11 @@ class NetworkXUnbounded(NetworkXAlgorithmError):
 
 class NetworkXNotImplemented(NetworkXException):
     """Exception raised by algorithms not implemented for a type of graph."""
+
+class InvalidNode(NetworkXException):
+    """Exception raised by an algorithm when source or target nodes are not
```

Comment:
This could be more general than a missing "source or target" node in an algorithm that expects a source or target node. It could be raised anywhere a node is expected to be found but is not in the graph. The docstring should reflect that.